### PR TITLE
Updates for local development

### PR DESCRIPTION
**Pull request by rebkwok**
Some updates for getting a local development server running

- added a `read_env()` to settings.py so it'll read env variables from the file if it exists
- updated README to use the dev-env.sh script for native and docker setup
- a couple of updates to the dev-env.sh script to get it working for me (ubuntu 20.04)
- Changed the Testing section to document how to run tests (it was duplicating the docker-compose section)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,23 +24,34 @@ Errors are logged to the DataLab [Sentry](https://sentry.io) account.
 
 ## Local Development Set Up
 ### Native
+
+Prerequisites:
+- **Python v3.9.x**
+- **Pip**
```
Comment (ghickman) [inline README.md:30]: Nit: probably virtualenv too?  I'm not sure all pythons come with `venv` installed.

File: README.md [hunk 2]
```diff
@@ -70,7 +88,9 @@ The DataLab job server is deployed to our `dokku2` instance, instructions are ar
 Run `make dev-config` if you have not already. Note: you will need the
 bitwarden cli tool installed in order to access passwords.
 
-A Django development server can be started with `docker-compose up`.
+Run the tests with:
+
+    pytest tests
```
Comment (ghickman) [inline README.md:93]: There's a makefile target for this, `test`, which runs both coverage and collectstatic.  It's what CI uses and I think we should push for that one by default as it's what's required to get the green ticks for CI.

File: scripts/dev-env.sh [hunk 3]
```diff
@@ -22,9 +24,9 @@ GH_DEV_TOKEN_BW_ID=10242708-234a-4162-982f-ace700ee5c53
 write() {
     local name="$1"
     local value="$2"
-    sed -i "s/$name=.*/$name=$value/" "$target"
+    sed -i"" -e "s/$name=.*/$name=$value/" "$target"
```
Comment (ghickman) [inline scripts/dev-env.sh:27]: Have you checked this is portable between macOS (which uses BSD sed) and Linux?

**Review by ghickman — APPROVED** [review]
Couple of nits but otherwise great 👍

File: scripts/dev-env.sh (hunk 3 above)
Comment (rebkwok) [reply scripts/dev-env.sh:27]: I'm on linux now and it works

**Comment by rebkwok** [comment]
Changed a couple of other commands for `make` ones (`make setup`, `make run`) as well as `make test` where they already existed

**Pull request merged**